### PR TITLE
[All hosts] (JS) Replace var with let/const in JS samples

### PR DESCRIPTION
**Pull request by samantharamon**
Updated the following:
- docs/code-snippets/*
- manifest/functionfile.md
- requirement-sets/outlook/*

### CODE REVIEW

File: generate-docs/json/office/snippets.yaml [hunk 1]
```diff
@@ -314,7 +314,7 @@
 
     // return the data selected in the document:
 
-    var dataValue = result.value;
+    // const dataValue = result.value;
```
Comment (alison-mk) [inline generate-docs/json/office/snippets.yaml:317]: Same question as other instances – should this be commented out?

File: generate-docs/json/office_release/snippets.yaml [hunk 2]
```diff
@@ -314,7 +314,7 @@
 
     // return the data selected in the document:
 
-    var dataValue = result.value;
+    // const dataValue = result.value;
```
Comment (alison-mk) [inline generate-docs/json/office_release/snippets.yaml:317]: Same question

**Review by alison-mk — APPROVED** [review]
Looks good to me, thanks for making all these changes! Approving with a couple comments (asking the same question in 4 locations).

File: docs/docs-ref-autogen/office/office/office.asyncresult.yml [hunk 3]
```diff
@@ -36,7 +36,7 @@ remarks: |-
   // When the call to the getSelectedDataAsync method completes, the callback function executes, 
   // and the following line of code accesses the value property of the AsyncResult object to 
   // return the data selected in the document:
-  var dataValue = result.value;
+  // const dataValue = result.value;
```
[the remark this answers is not in the document]
Comment (samantharamon) [reply docs/docs-ref-autogen/office/office/office.asyncresult.yml:39]: Thanks again for reviewing this, @alison-mk! Yes, this line is intentionally commented out as it only serves as a reference within the comment block. The variable is declared in a function a few lines above the comment block.

**Pull request merged**